sstables/compaction.cc

```diff
@@ -725,21 +727,19 @@ class cleanup_compaction final : public regular_compaction {
         clogger.info("Cleaned {}", formatted_msg);
     }
 
-    std::function<bool(const dht::decorated_key&)> filter_func() const override {
+    virtual bool filter_func(const dht::decorated_key& dk) const override {
         dht::token_range_vector owned_ranges = service::get_local_storage_service().get_local_ranges(_schema->ks_name());
 
-        return [this, owned_ranges = std::move(owned_ranges)] (const dht::decorated_key& dk) {
-            if (dht::shard_of(dk.token()) != engine().cpu_id()) {
-                clogger.trace("Token {} does not belong to CPU {}, skipping", dk.token(), engine().cpu_id());
-                return false;
-            }
+        if (dht::shard_of(dk.token()) != engine().cpu_id()) {
+            clogger.trace("Token {} does not belong to CPU {}, skipping", dk.token(), engine().cpu_id());
+            return false;
+        }
```

Comment:
This code reindenting should have been done in a separate patch.

Reply:
Will fix in v3.